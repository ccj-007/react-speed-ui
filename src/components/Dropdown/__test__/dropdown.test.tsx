import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Dropdown, Button } from '../../index';

const menu = [
  {
    key: '1',
    label: <div>我是默认的</div>,
  },
  {
    key: '2',
    label: (
      <a target='_blank' rel='noopener noreferrer' href='https://www.aliyun.com'>
        2nd menu item
      </a>
    ),
  },
  {
    key: '3',
    label: (
      <a target='_blank' rel='noopener noreferrer' href='https://www.luohanacademy.com'>
        3rd menu item
      </a>
    ),
  },
];

describe('test Dropdown component', () => {
  it('should render', () => {
    const { asFragment } = render(
      <Dropdown menu={menu}>
        <Button>打开</Button>
      </Dropdown>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
