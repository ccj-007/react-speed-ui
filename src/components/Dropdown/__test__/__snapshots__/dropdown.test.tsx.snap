// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Dropdown component should render 1`] = `
<DocumentFragment>
  <button
    class="speed-button speed-dropdown-warp speed-button-primary css-1dis8g1-Button"
    data-testid="button"
  >
    <div>
      <div />
      <div
        class="speed-menu speed-dropdown-b"
      />
    </div>
  </button>
</DocumentFragment>
`;
